mono/metadata/w32file-unix.c

```diff
@@ -4885,6 +4912,10 @@ mono_w32file_init (void)
 	finds = g_hash_table_new_full (g_direct_hash, g_direct_equal, NULL, finds_remove);
 	mono_coop_mutex_init (&finds_mutex);
 
+#if __APPLE__
+	clonefile_ptr = (clonefile_fn)dlsym (RTLD_DEFAULT, "clonefile");
```

Comment:
AFAICT this won't work on iOS or watchOS. @rolfbjarne can you give us a more authoritative answer here?

Reply:
We must not use RTLD_DEFAULT as it will cause us all sorts of trouble with embeders and fullAOT scenarios.

Reply:
I'd appreciate any suggestions on how to fix it. I assumed RTLD_DEFAULT is ok since it's a system API (libc, or libsystem respectively).

Reply:
Something like
```c
libc_handle = dlopen("/usr/lib/libc.dylib", RTLD_NOW);
clonefile_ptr = (clonefile_fn)dlsym (libc_handle, "clonefile");
```
?
How about iOS? I'd be fine with supporting macOS only if `dlsym` is a problem.